File: frontend/src/types/rag.ts
export interface SourceChunk {
  document_id: string | null;
  document_name: string | null;
  source_type: string;
  snippet: string;
  relevance_score: number;
}

export interface RagQueryResponse {
  id: string;
  question: string;
  answer: string;
  sources: SourceChunk[];
  model_used: string;
  processing_time_ms: number;
}

export interface RagStats {
  total_embeddings: number;
  total_documents_ingested: number;
  total_sop_chunks: number;
  total_queries: number;
}

export interface ChatMessage {
  role: "user" | "assistant";
  content: string;
  sources?: SourceChunk[];
  processing_time_ms?: number;
  timestamp: string;
}

File: backend/app/schemas/rag.py
"""Pydantic schemas for RAG Q&A API requests and responses."""

from uuid import UUID

from pydantic import BaseModel, Field


class RagQueryRequest(BaseModel):
    question: str = Field(..., min_length=3, max_length=2000)


class SourceChunk(BaseModel):
    document_id: UUID | None = None
    document_name: str | None = None
    source_type: str
    snippet: str
    relevance_score: float


class RagQueryResponse(BaseModel):
    id: UUID
    question: str
    answer: str
    sources: list[SourceChunk] = []
    model_used: str
    processing_time_ms: int


class RagStatsResponse(BaseModel):
    total_embeddings: int
    total_documents_ingested: int
    total_sop_chunks: int
    total_queries: int
